spec/models/prison_spec.rb

```diff
@@ -0,0 +1,166 @@
+      let(:expected_dates) { Set.new [Date.new(2015, 7, 29), Date.new(2015, 12, 25)] }
+
+      it 'returns a set containing the dates' do
+        expect(subject.unbookable_dates).to eq expected_dates
+      end
+    end
+    context 'when a prison does not have unbookable dates' do
+      subject { constructor_for mock_prison_data.except 'unbookable' }
+
+      it 'returns an empty set' do
+        subject.unbookable_dates.tap do |set|
+          expect(set).to be_empty
+          expect(set).to be_kind_of Set
+        end
+      end
+    end
+  end
+
+  describe '.visiting_slots' do
+    it 'returns the prisons slots as a hash' do
+      expect(subject.visiting_slots).to eq mock_slots_data
+    end
+  end
+
+  describe '.visiting_slot_days' do
+    let(:expected_slot_days) { %w<mon tue wed thu fri sat sun> }
+    it 'returns the abbreviated day names of available visiting days' do
+      expect(subject.visiting_slot_days).to eq expected_slot_days
+    end
+  end
+
+  describe '.anomalous_dates' do
+    context 'when a prison has days for visitation that are different from the regular slots' do
+      let(:expected_anomalous_dates) { Set.new [Date.new(2015, 8, 14)] }
+      it 'returns the day' do
+        expect(subject.anomalous_dates).to eq expected_anomalous_dates
+      end
+    end
+    context 'when a prison has no anomalous slots' do
+      subject { constructor_for mock_prison_data.except 'slot_anomalies' }
+
+      it 'returns an empty set' do
+        subject.anomalous_dates.tap do |set|
+          expect(set).to be_empty
+          expect(set).to be_kind_of Set
+        end
+      end
+    end
+  end
+
+  describe '.days_lead_time' do
+    context 'when a prison has a lead time explicitly set' do
+      it 'returns the value set by the prison config' do
+        expect(subject.days_lead_time).to eq 4
+      end
+    end
+    context 'when no lead time has been set' do
+      subject { constructor_for mock_prison_data.except 'lead_days' }
+
+      it 'returns the default value' do
+        expect(subject.days_lead_time).to eq 3
+      end
+    end
+  end
+
+  describe '.booking_window' do
+    context 'when a prison has a booking window explicitly set' do
+      it 'returns the value set by the prison config' do
+        expect(subject.booking_window).to eq 14
+      end
+    end
+    context 'when no booking window has been set' do
+      subject { constructor_for mock_prison_data.except 'booking_window' }
+
+      it 'returns the default value' do
+        expect(subject.booking_window).to eq 28
+      end
+    end
+  end
+
+  describe '.works_weekends?' do
+    context 'when a prison has a works weekends boolean flag set' do
+      it 'returns that boolean' do
+        expect(subject.works_weekends?).to be_truthy
+      end
+    end
+    context 'when no booking window has been set' do
+      subject { constructor_for mock_prison_data.except 'works_weekends' }
+
+      specify{ expect(subject.works_weekends?).to be_falsey }
```

Comment:
This is purely for the output it produces when the tests are run.

<img width="390" alt="screen shot 2015-07-17 at 10 22 25" src="https://cloud.githubusercontent.com/assets/1006365/8744434/13af1104-2c6f-11e5-8b68-f6f6a78cae0a.png">

I think within the context of the spec itself & reading the code its still sensible. What do you think?